src/main/java/com/salesforce/dynamodbv2/mt/repo/MtDynamoDbTableDescriptionRepo.java

```diff
@@ -293,25 +295,53 @@ private String getPrefix() {
         ScanRequest scanReq = new ScanRequest(getTableDescriptionTableName());
-        Map<String, AttributeValue> lastEvaluatedKey =
-            Optional.ofNullable(listMetadataRequest.getExclusiveStartTableMetadata())
-                .map(t -> new HashMap<>(ImmutableMap.of(tableDescriptionTableHashKeyField,
-                    new AttributeValue(getHashKey(t)))))
-                .orElse(null);
-        ScanResult scanResult;
-
-        scanReq.setExclusiveStartKey(lastEvaluatedKey);
-        scanReq.setLimit(listMetadataRequest.getLimit());
-        scanResult = amazonDynamoDb.scan(scanReq);
-        List<MtCreateTableRequest> createTableRequests = scanResult.getItems().stream()
-            .map(rowMap ->
-                new MtCreateTableRequest(
-                    getTenantTableFromHashKey(rowMap.get(tableDescriptionTableHashKeyField).getS()).getTenantName(),
-                    getCreateTableRequest(jsonToTableData(rowMap.get(tableDescriptionTableDataField).getS()))))
-            .collect(Collectors.toList());
-        MtCreateTableRequest lastEvaluatedMetadata = scanResult.getLastEvaluatedKey() == null ? null :
-            createTableRequests.get(createTableRequests.size() - 1);
-        return new ListMetadataResult(createTableRequests, lastEvaluatedMetadata);
+        Map<String, AttributeValue> lastEvaluatedKey = null;
+        if (listMetadataRequest.getStartTenantTableKey() != null) {
+            lastEvaluatedKey = new HashMap<>(ImmutableMap.of(tableDescriptionTableHashKeyField,
+                new AttributeValue(getHashKey(listMetadataRequest.getStartTenantTableKey()))));
+        }
+        // if multitenant virtual table list, this is easy, scan for the rows up to limit, and returned what's found
+        if (mtContext.getContextOpt().isEmpty()) {
+            ScanResult scanResult;
+
+            scanReq.setExclusiveStartKey(lastEvaluatedKey);
+            scanReq.setLimit(listMetadataRequest.getLimit());
+            scanResult = amazonDynamoDb.scan(scanReq);
+            List<MtCreateTableRequest> createTableRequests = scanResult.getItems().stream()
+                .map(rowMap ->
+                    new MtCreateTableRequest(
+                        getTenantTableFromHashKey(rowMap.get(tableDescriptionTableHashKeyField).getS()).getTenantName(),
+                        getCreateTableRequest(jsonToTableData(rowMap.get(tableDescriptionTableDataField).getS()))))
+                .collect(Collectors.toList());
+            MtCreateTableRequest lastEvaluatedMetadata = scanResult.getLastEvaluatedKey() == null ? null :
+                createTableRequests.get(createTableRequests.size() - 1);
+            return new ListMetadataResult(createTableRequests, lastEvaluatedMetadata);
+        } else {
+            // if tenant scoped virtual table list, keep scanning until we either run out of metadata rows in the
+            // multitenant virtual table definitions, or we fill up our result set
```

Comment:
We should add a todo to our schema change backlog to make this a multi-tenant, hash-partitioned table, so that we can scan it efficiently. It's essentially the mt-dynamo equivalent of the metadata table in ZOS.